packages/grid/_modules_/grid/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -149,21 +145,15 @@ export function useGridDimensions(
   );
 
   const getViewportPageSize = React.useCallback(() => {
-    const dimensions = apiRef.current.getRootDimensions();
-
-    if (!dimensions) {
-      return 0;
-    }
+    const renderContext = apiRef.current.unstable_getRenderContext();
```

Comment:
I'm not a huge fan of exposing virtualization internals upper in the DataGrid component tree.
But I don't have a good solution either.

If @m4theushw has a better idea.
Otherwise, I'm fine with merging as is and taking time later to improve.

Reply:
Actually the keyboard navigation won't work as expected if `disableVirtualization=true`. With no virtualization, the last row index in the render context points to the last row in the entire row set. I'm waiting for #3667 to be merged to able to test if there're no regressions. In the worst scenario, I would keep the old behavior if `getRowHeight` is not being used.

Reply:
I tried calculating it without the `renderContext` -> the problem is that because of the variable row height you need to know the indexes of the first and last rows in the viewport to calculate the total number of visible rows. If anyone have any ideas I'm open to them.

Reply:
There's a regression here. Pressing <kbd>Page Up</kbd> is jumping directly to the column header while not on the first row yet. I don't mind if we ship this small bug only affecting those users using `getRowHeight`, but keeping the old behavior for those who don't use it. I think the correct way would be to base on `rowsMeta.positions` and the current scroll position.

<img width="600" src="https://user-images.githubusercontent.com/42154031/150799507-13ffbe9c-1b8c-4087-812a-6a7bd68649ef.gif" alt="msedge_srHruiStdO" style="max-width: 100%;">

Reply:
Updated!